Content.Server/GameTicking/Rules/ThiefRuleSystem.cs

```diff
@@ -0,0 +1,194 @@
+using Content.Server.Chat.Managers;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Mind;
+using Content.Server.Objectives;
+using Content.Server.Shuttles.Components;
+using Content.Server.Roles;
+using Content.Shared.Mind;
+using Content.Shared.Objectives.Components;
+using Content.Shared.Preferences;
+using Content.Shared.Roles;
+using Content.Shared.Roles.Jobs;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using Robust.Shared.Prototypes;
+using System.Linq;
+using Content.Shared.Humanoid;
+using Content.Server.Antag;
+using Robust.Server.Audio;
+
+namespace Content.Server.GameTicking.Rules;
+
+public sealed class ThiefRuleSystem : GameRuleSystem<ThiefRuleComponent>
+{
+    [Dependency] private readonly IChatManager _chatManager = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly AntagSelectionSystem _antagSelection = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly MindSystem _mindSystem = default!;
+    [Dependency] private readonly SharedRoleSystem _roleSystem = default!;
+    [Dependency] private readonly SharedJobSystem _jobs = default!;
+    [Dependency] private readonly ObjectivesSystem _objectives = default!;
+
+    const string bigObjectiveGroup = "ThiefBigObjectiveGroups";
+    const string smallObjectiveGroup = "ThiefObjectiveGroups";
+    const string escapeObjectiveGroup = "ThiefEscapeObjectiveGroups";
+
+    private const float BigObjectiveChance = 0.7f;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RulePlayerJobsAssignedEvent>(OnPlayersSpawned);
+
+        SubscribeLocalEvent<ThiefRoleComponent, GetBriefingEvent>(OnGetBriefing);
+        SubscribeLocalEvent<ThiefRuleComponent, ObjectivesTextGetInfoEvent>(OnObjectivesTextGetInfo);
+    }
+
+    private void OnPlayersSpawned(RulePlayerJobsAssignedEvent ev)
+    {
+        var query = EntityQueryEnumerator<ThiefRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var thief, out var gameRule))
+        {
+            //Chance to not lauch gamerule
+            if (!_random.Prob(thief.RuleChance))
+            {
+                RemComp<ThiefRuleComponent>(uid);
+                continue;
+            }
+
+            if (!GameTicker.IsGameRuleAdded(uid, gameRule))
+                continue;
+
+            foreach (var player in ev.Players)
+            {
+                if (!ev.Profiles.ContainsKey(player.UserId))
+                    continue;
+
+                thief.StartCandidates[player] = ev.Profiles[player.UserId];
+            }
+            DoThiefStart(thief);
+        }
```

Comment:
Use TryGetValue as it avoids indexing ev.Profiles twice.

Reply:
I'm sorry I didn't understand what was required